NebulaNetwork/PacketProcessors/Factory/Turret/TurretStorageUpdateProcessor.cs

```diff
@@ -0,0 +1,40 @@
+﻿#region
+
+using NebulaAPI.Packets;
+using NebulaModel.Networking;
+using NebulaModel.Packets;
+using NebulaModel.Packets.Factory.Ejector;
+using NebulaModel.Packets.Factory.Turret;
+
+#endregion
+
+namespace NebulaNetwork.PacketProcessors.Factory.Ejector;
+
+[RegisterPacketProcessor]
+internal class TurretStorageUpdateProcessor : PacketProcessor<TurretStorageUpdatePacket>
+{
+    protected override void ProcessPacket(TurretStorageUpdatePacket packet, NebulaConnection conn)
+    {
+        var pool = GameMain.galaxy.PlanetById(packet.PlanetId)?.factory?.defenseSystem.turrets;
+        if (pool == null || packet.TurretIndex == -1 || packet.TurretIndex > pool.buffer.Length ||
+            pool.buffer[packet.TurretIndex].id == -1)
+        {
+            return;
+        }
+
+        if (packet.ItemCount == 0 || packet.ItemId == 0)
+        {
+            pool.buffer[packet.TurretIndex].ClearItem();
```

Comment:
Would help a bit at runtime if this was a local variable. Avoids a memory load/store

Reply:
I followed the existing pattern of the app, all processors are done this way.  Any performance tuning is always a good thing, but code consistency, especially in an app with multiple maintainers, reigns king in my opinion. Well at least so long as the performance isn't an issue.